python/ray/_private/gcs_utils.py

```diff
@@ -100,17 +100,21 @@ def use_gcs_for_bootstrap():
 
 
 def get_gcs_address_from_redis(redis) -> str:
-    """Reads GCS address from redis.
+    """Reads GCS address from redis, with up to 10 retries.
```

Comment:
Why do we need to add retries in this PR for the legacy path?

Reply:
This path is updated during the recent refactoring for GCS redis removal, and it is called in more process start codepaths than before. I reverted the change and will put it back it we start to see flakiness.